skopt/optimizer/base.py

```diff
@@ -245,10 +245,8 @@ def base_minimize(func, dimensions, base_estimator,
     for n in range(n_calls):
         next_x = optimizer.ask()
 
-        # no need to fit a model on the last iteration
-        fit_model = n < n_calls - 1
         next_y = func(next_x)
-        result = optimizer.tell(next_x, next_y, fit=fit_model)
+        result = optimizer.tell(next_x, next_y)
```

Comment:
What was the thinking (other than performance) for not fitting a final model? Changed this because `expected_minimum()` and other users would expect that `res.models[-1]` is a model that includes all points. Without this fix you get the following:

![image](https://user-images.githubusercontent.com/1448859/30309634-96ef0a58-978c-11e7-8beb-bbcf9c06d5f3.png)

One of the green points isn't used for the model.

Reply:
i think it was only about performance.

Reply:
Let's change it. It confused the heck out of me making demo notebooks :-/